src/main/java/org/apache/sysds/runtime/matrix/data/MatrixBlock.java

```diff
@@ -1390,9 +1364,10 @@ public void copy(MatrixValue thatValue) {
 	public void copy(MatrixValue thatValue, boolean sp) 
 	{
 		MatrixBlock that = checkType(thatValue);
-		
 		if( this == that ) //prevent data loss (e.g., on sparse-dense conversion)
 			throw new RuntimeException( "Copy must not overwrite itself!" );
+		if(that instanceof CompressedMatrixBlock)
```

Comment:
As discussed this is not very nice to have all these references to CompressedMatrixBlock in the base class MatrixBlock. However, if CompressedMatrixBlock is indeed moved into MatrixBlock it might be ok.